providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
1: This is more of less pseudo-code. I somehow tested it, and it will work, but you know...
How could you ever expand a .* without checking the actual FS?
2: If we'd use -R we could drop the find all along. ;)

I think we have a plan. I'll try to get that ready within the next few hours.